stock_no_negative/models/stock_location.py

```diff
@@ -12,3 +12,7 @@ class StockLocation(models.Model):
         string='Allow Negative Stock',
         help="Allow negative stock levels for the stockable products "
         "attached to this location.")
+
+    @api.multi
+    def _is_negative_allowed(self):
+        return self.allow_negative_stock or self.usage not in ['internal', 'transit']
```

Comment:
Why or?

it should be AND because the old code 
`quant.location_id.usage in ['internal', 'transit'] and disallowed_by_product and disallowed_by_location`

Reply:
@hugho-ad it's the opposite code. The one you reference is to check if it's disallowed, and here is to determine if it's allowed.